workspaces/analytics/src/eventbus.ts

```diff
@@ -2,32 +2,36 @@ import { EventEmitter } from 'events';
 import { v4 as uuidv4 } from 'uuid';
 
 import {
-  ClientContext,
+  CliClientContext,
   TrackingEventBase,
 } from './interfaces/TrackingEventBase';
 
 export function newAnalyticsEventBus(
-  getContext: (batchId: string) => Promise<ClientContext>
+  getContext: (batchId: string) => Promise<CliClientContext>
 ): AnalyticsEventBus {
   const eventEmitter = new EventEmitter();
   return {
     emit: async (...events: TrackingEventBase<any>[]) => {
       const batchId = uuidv4();
       const context = await getContext(batchId);
-      // TODO - don't overwrite context
       events.forEach((event) => {
-        eventEmitter.emit('event', { ...event, context });
+        eventEmitter.emit('event', { event, context });
       });
     },
     eventEmitter,
-    listen: (callback: (e: TrackingEventBase<any>) => void): void => {
+    listen: (callback: ListenCallbackType): void => {
```

Comment:
This is probably the part that requires the most attention, since I don't think typescript can infer through events in the event emitter (it just trusts the events we give it)